internal/mongodoc/resource.go

```diff
@@ -0,0 +1,188 @@
+// Copyright 2016 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package mongodoc
+
+import (
+	"fmt"
+	"sort"
+	"time"
+
+	"gopkg.in/errgo.v1"
+	"gopkg.in/juju/charm.v6-unstable"
+	"gopkg.in/juju/charm.v6-unstable/resource"
+	"gopkg.in/juju/charmrepo.v2-unstable/csclient/params"
+	"gopkg.in/mgo.v2/bson"
+)
+
+// NewResourceQuery formats the provided details into a mongo query for
```

Comment:
Fair enough.  I was trying to keep code that has to refer to the BSON field names next to the doc, but I'll stick to your guidelines.